src/cmd/fontsrv/x11.c

```diff
@@ -59,44 +59,31 @@ load(XFont *f)
 {
 	FT_Face face;
 	FT_Error e;
-	FT_ULong charcode;
-	FT_UInt glyph_index;
+	int i;
 
 	if(f->loaded)
 		return;
 
 	e = FT_New_Face(lib, f->fontfile, f->index, &face);
-
 	if(e){
 		fprint(2, "load failed for %s (%s) index:%d\n", f->name, f->fontfile, f->index);
 		return;
 	}
-
 	if(!FT_IS_SCALABLE(face)) {
 		fprint(2, "%s is a non scalable font, skipping\n", f->name);
 		FT_Done_Face(face);
-	    f->loaded = 1;
+		f->loaded = 1;
 		return;
 	}
-
 	f->unit = face->units_per_EM;
 	f->height = (int)((face->ascender - face->descender) * 1.2);
 	f->originy = face->descender; // bbox.yMin (or descender)  is negative, becase the baseline is y-coord 0
 
-	for(charcode=FT_Get_First_Char(face, &glyph_index); glyph_index != 0;
-		charcode=FT_Get_Next_Char(face, charcode, &glyph_index)) {
-
-		int idx = charcode/SubfontSize;
-
-		if(charcode > 0xffff)
-			break;
-
-		if(!f->range[idx]) {
-			f->range[idx] = 1;
-			f->nrange++;
-		}
+	// enable all Unicode ranges
```

Comment:
I'm not sure I understand this change, the original code would scan the glyphs in the font and mark the present ones. Now you mark all as present, even if they are not?

https://www.freetype.org/freetype2/docs/reference/ft2-base_interface.html#FT_Get_Next_Char

Reply:
`libdraw` expects U+0000 to be present. Initially, I had the code make sure U+0000 was always present, but then I saw OS X enables all ranges, so I did the same. Since we now make sure empty subfonts don't cause any problem, it doesn't hurt to enable everything.

Reply:
I think the OSX code was changed because it used `CGFontGetGlyphsForUnichars` which is not available anymore(?), I don't feel very strongly about it so +1 and thanks again!

Reply:
OSX enabling all the ranges is because it fills in the missing glyphs with glyphs from a set of default fonts.  Do you think you can make freetype do it?  Or do we need fontconfig for backup font families?